pkg/cableengine/healthchecker/healthchecker.go

```diff
@@ -116,6 +120,19 @@ func (h *controller) Start(stopCh <-chan struct{}) error {
 	logger.Infof("CableEngine HealthChecker started with PingInterval: %v, MaxPacketLossCount: %v", h.config.PingInterval,
 		h.config.MaxPacketLossCount)
 
+	// Start the Pinger for the endpoints that is already present.
+	endpoints, err := h.config.SubmarinerClient.SubmarinerV1().Endpoints(h.config.EndpointNamespace).
+		List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing endpoints")
+	}
+
+	endpointList := endpoints.Items
+
+	for i := range endpointList {
+		h.endpointCreatedOrUpdated(&endpointList[i], 0)
+	}
+
```

Comment:
So this gets all `Endpoints` and runs them through `endpointCreatedOrUpdated`. This duplicates what the informer already does so I assume this is to handle subsequent calls to `Start`. While this will reprocess current `Endpoints`, the `endpointWatcher` will not be restarted so it will not handle updates to remote `Endpoints`.

The real problem is that the `controller` is not re-startable and the caller assumes it is. To make it so, we need to create the `endpointWatcher` instance in the `Start` method so it's recreated on each call.